unified-scheduler-pool/src/lib.rs

```diff
@@ -1115,6 +1231,15 @@ impl<S: SpawnableScheduler<TH>, TH: TaskHandler> ThreadManager<S, TH> {
         mut result_with_timings: ResultWithTimings,
         handler_context: HandlerContext,
     ) {
+        let scheduling_mode = context.mode();
+        let (mut is_finished, mut session_ending) = match scheduling_mode {
+            BlockVerification => (false, false),
+            BlockProduction => {
+                assert!(context.is_preallocated());
+                (true, true)
+            }
+        };
+
```

Comment:
I can't get to the line in review.
There's a series of comments on line 1332:
```rust
        // High-level flow of new tasks:
        // 1. the replay stage thread send a new task.
        // 2. the scheduler thread accepts the task.
        // 3. the scheduler thread dispatches the task after proper locking.
        // 4. the handler thread processes the dispatched task.
        // 5. the handler thread reply back to the scheduler thread as an executed task.
        // 6. the scheduler thread post-processes the executed task.
```

we should update this comment imo, but it doesn't necessarily have to be part of this PR - maybe not until the full implementation of production w/ all verification is in.

Reply:
yeah, I'll update them later at once.